docs/hip-16.md

```diff
@@ -0,0 +1,96 @@
+---
+hip: 16
+title: Entity Auto Renewal
+author: Leemon Baird (@lbaird), Nosh Mody (@noshmody), Quan Nguyen (@qnswirlds)
+type: Standards Track
+category: Service
+status: Draft
+created: 2021-03-29
+discussions-to: <a URL pointing to the official discussion thread>
+updated:
+requires:
+replaces:
+superseded-by:
+---
+
+## Abstract
+
+When a Hedera entity is created, the payer account is charged enough hbars (as a rental fee) for the entity to stay active
+in the ledger state until consensus time passes its _expiration time_. Users can extend the expiration time of an entity by
+paying an extension fee via an update transaction. This HIP defines and discusses another mechanism to be implemented by
+Hedera Services to automatically renew expired entities using funds of linked _autorenew accounts_ or _admin accounts_; and
+automatically remove expired entities that lack a funded autorenew account (or are deleted).
+
+## Motivation
+
+Prior to this HIP, the expiration time of a Hedera entity has not been checked or enforced. An entity remains active in
+the ledger even after its expiration time, without additional fees being charged. Upon implementation of this HIP,
+Hedera Services will __begin to charge rent__ for automatically renewed entities; and will remove from the ledger expired
+entities which are either deleted, or have an admin/autorenew account with zero balance at the time renewal fees are due.
+
+## Rationale
+
+This section seems to be a duplicate of the `Motivation` section above. We will add more details if required.
+
+## Specification
+
+### Terminologies
+- Deletion - A successful delete transaction will mark an entity as deleted and that entity cannot be operated up on.
+The entity will remain in the ledger, marked as deleted, until it expires.
```

Comment:
Does the deletion transaction also update the expiration time to a closer date? e.g. `now() + min('expiration_time','grace_period')`

If not waiting till the expiration time would result in entities that don't need to be in state being kept around. This makes sense in the regular case where renewal could occur, but an explicit deletion should probably ensure deletion is done sooner rather than later.